rblib/format.rb

```diff
@@ -60,8 +60,8 @@ def self.make_clickable(text, params = {})
             # http://www.whatdotheyknow.com/request/advice_sought_from_information_c#incoming-24711
             ret = ret.gsub(/( LTCODE http[\S\r\n]*? GTCODE )/) { |m| m.gsub(/[\n\r]/, "") }
 
-            ret = ret.gsub(/(https?):\/\/([^\s<>{}()]+[^\s.,<>{}()])/i, "<a href='\\1://\\2'" + (nofollow ? " rel='nofollow'" : "") + ">\\1://\\2</a>")
-            ret = ret.gsub(/(\s)www\.([a-z0-9\-]+)((?:\.[a-z0-9\-\~]+)+)((?:\/[^ <>{}()\n\r]*[^., <>{}()\n\r])?)/i,
+            ret = ret.gsub(/(https?):\/\/([^\s<>]+[^\s.,<>])/i, "<a href='\\1://\\2'" + (nofollow ? " rel='nofollow'" : "") + ">\\1://\\2</a>")
+            ret = ret.gsub(/(\s)www\.([a-z0-9\-]+)((?:\.[a-z0-9\-\~]+)+)((?:\/[^ <>\n\r]*[^., <>\n\r])?)/i,
```

Comment:
I don't know of any method to address this problem. When I enter a URL and expect it to be rendered into a clickable URL, I add a space at the end to make sure there is no confusion. A . or , or ) is not common at the end of a URL, therefore it would make sense to exclude them. The problem is that if a URL has a trailing [,.)] the user has no way to fix this. If the trailing [.,)] are not treated specially, the user still has the option of adding a space to avoid adding the character to the URL. I would advocate for not treating the trailing [,.)] in a special way for this reason. But it's probably the topic of another pull request that would rework the logic of URL parsing ?